src/read.c

```diff
@@ -3958,8 +3923,53 @@ static avifResult avifDecoderDecodeTiles(avifDecoder * decoder,
             }
         }
 
-        ++*decodedTileCount;
+        ++tiles->decodedTileCount;
+    }
+    return AVIF_RESULT_OK;
+}
+
+avifResult avifDecoderOutputDecodedTiles(avifDecoder * decoder, const avifItemTiles * tiles, avifBool alpha, unsigned int oldDecodedTileCount)
+{
+    if (tiles->decodedTileCount == oldDecodedTileCount) {
+        // Nothing changed.
+        return AVIF_RESULT_OK;
+    }
+    // There is at least one newly decoded tile.
+    assert(tiles->decodedTileCount > oldDecodedTileCount);
+
+    // Grid path.
+    if ((tiles->grid.rows > 0) && (tiles->grid.columns > 0)) {
+        assert(tiles->tileCount == (tiles->grid.rows * tiles->grid.columns));
+        AVIF_CHECKERR(avifDecoderDataFillImageGrid(decoder->data, decoder->image, tiles, oldDecodedTileCount, alpha),
+                      AVIF_RESULT_INVALID_IMAGE_GRID);
+        return AVIF_RESULT_OK;
+    }
+
+    // Normal (most common) non-grid path.
```

Comment:
Nit/optional: Should we preserve the full comment from the original code?

```
// Normal (most common) non-grid path. Just steal the planes from the only "tile".
```

Either is fine by me. Just wanted to confirm you meant to delete the second sentence in the original comment.